js/classes/SearchResult.js

```diff
@@ -0,0 +1,297 @@
+
+            title.innerHTML = this.highlightSearchText(
+                this.formatSymbol(element.symbol)
+            );
+
+            title.className = "card-title";
+            title.setAttribute("color", "red");
+
+            let subtitle = document.createElement("h6");
+            subtitle.innerHTML = this.highlightSearchText(element.name);
+
+            subtitle.className = "card-subtitle";
+
+            cardBody.appendChild(title);
+            cardBody.appendChild(subtitle);
+            titleURL.appendChild(cardBody);
+            card.appendChild(titleURL);
+            cards.push(card);
+        }
+
+        let symbols = data.map((element) => element.symbol);
+
+        let limit = Math.min(cards.length, API_PROFILE_COMPANY_LIMIT);
+        let symbolsFirstThree = symbols.slice(0, limit);
+
+        let symbolsString = symbolsFirstThree.join(",");
+
+        // fix for one result to keep response format consistent when requesting
+        if (symbolsFirstThree.length === 1) {
+            symbolsString += ",";
+        }
+
+        //fetching first three symbols (max qty) to get company profile.
+        this.getCompanyProfiles(symbolsString, symbolsFirstThree, cards);
+
+        return cards;
+    };
+
+    getPromiseSearchData = () => {
+        let data = fetch(this.url)
+            .then((response) => {
+                if (response.ok) {
+                    return response.json();
+                } else {
+                    throw new Error(
+                        "Network response error: " + response.status
+                    );
+                }
+            })
+            .then((data) => {
+                this.hideLoading();
+                if (typeof data === "object" && data.length > 0) {
+                    let cards = [];
+                    cards = this.createStockCards(data);
+                    this.element.replaceChildren(...cards);
+                } else {
+                    this.element.replaceChildren();
+                }
+            })
+            .catch(function (data) {
+                console.error(data);
+                this.element.replaceChildren();
+            });
+
+        return data;
+    };
+
+    formatSymbol = (symbol) => {
+        let symbolFormatted = "<b>" + SYMBOL_PREFIX + symbol + "</b>";
+        return symbolFormatted;
+    };
+
+    createSearchThumbnail = (src) => {
+        let img = document.createElement("img");
+        img.src = src;
+        img.onerror = function () {
+            img.src = "./img/stocki-logo-simple.png";
+        };
+        img.className = "card-img-top";
+        img.classList.add("img"), (img.style.width = "100px");
+
+        return img;
+    };
+
+    /* set a caret and a trend color for stock percentages */
+    setTrendColor = (trend, changes) => {
+        trend.className = changes > 0 ? "fa fa-caret-up" : "fa fa-caret-down";
+    };
+
+    getCompanyProfiles(symbolsString, symbolsFirstThree, cards) {
```

Comment:
there is a bug in this function, you are getting the information obly for 3 companies.